plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -299,6 +455,39 @@ private Optional<DataFileInfo> rewriteParquetFile(Path path, ImmutableLongBitmap
         return Optional.of(fileWriter.getDataFileInfo());
     }
 
+    private void storeCdfEntries(Page page, int[] deleted, int deletedCount, FileDeletion deletion)
+    {
+        if (cdfEnabled) {
+            if (cdfPageSink == null) {
+                cdfPageSink = cdfPageSinkSupplier.get();
+            }
+            Page cdfPage = page.getPositions(deleted, 0, deletedCount);
+            Block[] outputBlocks = new Block[nonSynthesizedColumns.size() + 1];
+            int cdfPageIndex = 0;
+            int partitionIndex = 0;
+            List<String> partitionValues = deletion.partitionValues;
+            for (int i = 0; i < nonSynthesizedColumns.size(); i++) {
+                if (nonSynthesizedColumns.get(i).getColumnType() == REGULAR) {
+                    outputBlocks[i] = cdfPage.getBlock(cdfPageIndex);
+                    cdfPageIndex++;
+                }
+                else {
+                    outputBlocks[i] = RunLengthEncodedBlock.create(nativeValueToBlock(
+                            nonSynthesizedColumns.get(i).getType(),
+                            deserializePartitionValue(
+                                    nonSynthesizedColumns.get(i),
+                                    Optional.of(partitionValues.get(partitionIndex)))
```

Comment:
Can partition value be null?

Reply:
it can't it comes from `FileDeletion` and there is a specific check for nulls

Reply:
Does that mean we cannot delete from a null partition?
Do we have a test for that, with CDF enabled?

Reply:
ok I missunderstood, i meant that `partitionValues` can't be null, value returned by that get can be null - so it should be `Optional.ofNullable`

Reply:
makes sense, thanks for clarifying!
Do we have a test for that, with CDF enabled?